examples/tight_binding/tight_binding.F

```diff
@@ -0,0 +1,1229 @@
+                 do ib = jb,h ! SK integrals are even
+                    matrix_element = vpp_nondiag(ib-1, jb-1, vec_norm, Helem(t1, vppp), Helem(t1, vpps))
+                    values(ib + h * (jb-1)) = values(ib + h * (jb-1)) + matrix_element
+                    values(jb + h * (ib-1)) =  values(ib + h * (jb-1))
+                 enddo
+              enddo
+              
+           endif
+        endif
+      enddo ! irft       
+      
+      if (add_block) then
+         call dbcsr_put_block(ham, ii, jj, values(1:nze))
+      endif
+    end do
+
+    call dbcsr_get_stored_coordinates(ham, ii, ii, node_holds_blk)
+    
+    if (node_holds_blk .eq. mynode) then
+       w = row_blk_sizes(ii)
+       h = col_blk_sizes(ii)
+       nze = w * h
+       values(:) = 0
+       ! calc onsite
+       ! Es
+       values(1) = diagH(rho, energiesS)
+       diagVal = diagH(rho, energiesP)
+
+       ! Ep
+       do ib=2,w
+          values(ib + h * (ib - 1)) = diagVal
+       enddo
+       call dbcsr_put_block(ham, ii, ii, values(1:nze))
+    endif
+
+  end do 
+!$OMP END DO
+  DEALLOCATE (values)
+  
+!$OMP END PARALLEL
+  return
+end subroutine
+
+
+subroutine build_ham_blocked_dense_no_trans(ham, rr, nr, block_size, distance_cutoff)
+  implicit none
+  integer, intent(in) :: nr, block_size
+  real(8), intent(in) :: rr(3,nr), distance_cutoff
+  real(8), allocatable, intent(inout) :: ham(:,:)
+
+  real(8), parameter :: rcut=10.d0
+  integer ii,jj,irft, ib,jb, i,j
+  real(8) t1,z2
+  real(8) r01(2),r02(2),r12(2)
+  real(8) r1(3),r2(3),dvec(3)
+  integer info,n,ld 
+  integer lwork
+  real(8), allocatable :: rwork(:)
+  real(8), allocatable :: work(:)
+  real(8) :: vec_norm(3), matrix_element,diagVal, rho
+  integer :: nzz
+  integer :: h, w
+  
+  ham(:,:) = 0.0d0
+  nzz = 0
+   
+  do ii=1,nr
+    r1=rr(:,ii)
+    rho = 0
+    i = (ii - 1) * block_size + 1
+    
+    do jj=1,nr
+      r2=rr(:,jj)
+      dvec(:)=r2(:)-r1(:)
+      t1=sqrt(dvec(1)**2+dvec(2)**2+dvec(3)**2)
+      vec_norm(:) = dvec(:) / t1
+      
+      j = (jj - 1) * block_size + 1
+      
+      if (t1 < distance_cutoff .and.  ii /= jj) then
+         rho = rho + rhoPart(lambda, t1) 
+         
+         ! calc non-diag
+         ham(i,j) = Helem(t1, vsss)
+         ham(j,i) = ham(i,j)
+         do jb = 1,3
+            matrix_element = vsp_nondiag(jb, vec_norm, Helem(t1,vsps)) 
+            
+            ham(i, j+jb) =   matrix_element 
+            ham(i+jb, j) = - matrix_element
+            
+            
+            do ib = jb,3
+            matrix_element = vpp_nondiag(ib, jb, vec_norm, Helem(t1, vppp), Helem(t1, vpps))
+            ham(i+ib, j+jb) =  matrix_element
+            ham(i+jb, j+ib) =  matrix_element
+            enddo
+         enddo
+      endif
+      
+            
+      ! calc onsite
+      ! Es
+      ham(i,i) = diagH(rho, energiesS)
+      diagVal = diagH(rho, energiesP)
+
+      ! Ep
+      do ib= 1,3
+         ham(i + ib, i + ib ) = diagVal
```

Comment:
maybe `ham(i+1:i+3,i+1:i+3) = diagVal` instead?

Reply:
will it give me a matrix 3x3, not diagonal?